rocketpy/simulation/flight_data_importer.py

```diff
@@ -0,0 +1,399 @@
+    "alpha3": "Angular acceleration z (rad/s^2)",
+    "altitude": "Altitude AGL (m)",
+    "ax": "Acceleration x (m/s^2)",
+    "ay": "Acceleration y (m/s^2)",
+    "az": "Acceleration z (m/s^2)",
+    "bearing": "Bearing (deg)",
+    "density": "Density (kg/m^3)",
+    "drift": "Drift (deg)",
+    "e0": "Quaternion e0",
+    "e1": "Quaternion e1",
+    "e2": "Quaternion e2",
+    "e3": "Quaternion e3",
+    "latitude": "Latitude (deg)",
+    "longitude": "Longitude (deg)",
+    "mach_number": "Mach number",
+    "name": "Time (s)",
+    "speed": "Speed (m/s)",
+    "stability_margin": "Stability margin",
+    "static_margin": "Static margin",
+    "vz": "Velocity z (m/s)",
+    "vx": "Velocity x (m/s)",
+    "vy": "Velocity y (m/s)",
+    "w1": "Angular velocity x (rad/s)",
+    "w2": "Angular velocity y (rad/s)",
+    "w3": "Angular velocity z (rad/s)",
+    "x": "Position x (m)",
+    "y": "Position y (m)",
+    "z": "Position z (m)",
+}
+
+
+class FlightDataImporter:
+    """A class to create a rocketpy.Flight object from a .csv file. The data
+    frame must contain a time column in seconds.
+
+    There are different ways of calculating a property from the data frame.
+        1. The property is already in data frame and you want to use it as is
+        2. The property is not in the data frame but you can calculate it from
+            other properties in the data frame.
+
+    Initially, this class is designed to work with the option '1.'
+    """
+
+    def __init__(
+        self,
+        name,
+        paths,
+        columns_map=None,
+        units=None,
+        interpolation="linear",
+        extrapolation="zero",
+        separator=",",
+        encoding="utf-8",
+    ):
+        """Initializes the FlightDataImporter class.
+
+        Parameters
+        ----------
+        name : str
+            The name of the FlightDataImporter object.
+        paths : list
+            A list of paths or strings representing the file paths or directories
+            containing the flight data files. Only .csv and .txt files are supported.
+        columns_map : dict, optional
+            A dictionary mapping column names to desired column names.
+            Defaults to None, which will keep the original column names.
+        units : dict, optional
+            A dictionary mapping column names to desired units.
+            Defaults to None, which will consider that all the data is in SI.
+        interpolation : str, optional
+            The interpolation method to use for missing data.
+            Defaults to "linear", see rocketpy.mathutils.Function for more
+            information.
+        extrapolation : str, optional
+            The extrapolation method to use for data outside the range.
+            Defaults to "zero", see rocketpy.mathutils.Function for more
+            information.
+        separator : str, optional
+            The separator used in the data file. Defaults to ",".
```

Comment:
`delimiter` is a more common name for this

Reply:
Good catch. The numpy module also uses delimiter, so I've changed in my latest commit.